service/controller/resource/certificates/resource.go

```diff
@@ -117,10 +139,15 @@ func (r *Resource) getSource(ctx context.Context, v interface{}) (*corev1.Secret
 
 		secret, err = r.k8sClient.K8sClient().CoreV1().Secrets(secretNamespace).Get(ctx, secretName, metav1.GetOptions{})
 		if apierrors.IsNotFound(err) {
-			// fall through
+			secret = nil
```

Comment:
Why do we need this? I would expect `secret` to be `nil` when there is an error.

Reply:
I just do not really like empty methods :D

Reply:
Interestingly, this returns an empty secret by default:

https://github.com/kubernetes/client-go/blob/fa8f4cc307b81b125f6beb2cea6f093ed9e3c069/kubernetes/typed/core/v1/secret.go#L72